### PR TITLE
DM-7406: Create Light curve skeleton app

### PR DESCRIPTION
**Pull request by loitly**
- introduce template into firefly entry point.
 - create light curve as a template.
 - create viewer, controller, and results component for light curve.
   - these components are place holders and are not fully implemented.
 - create upload form for use during development.

To view light curver viewer use lc.html:  ie. http://localhost:8080/firefly/lc.html
Upload a table to see the results.

### CODE REVIEW

**Comment by ejoliet** [comment]
Looks great!!! Nice starter page! I've uploaded a file and checked that everything was ok.
LC layout and skeleton is good. Review done.

File: src/firefly/html/lc.html [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+<head>
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
+    <link rel="icon" type="image/x-icon" href="images/fftools-logo-16x16.png">
+    <meta name="apple-mobile-web-app-capable" content="yes" />
+    <title>Light Curve Viewer</title>
+
+    <script>
+        window.firefly = {
+            app: {
+                template: 'LightCurveViewer',
```
Comment (robyww) [inline src/firefly/html/lc.html]: The triview is the default template?  I think we should probably specify it.

File: src/firefly/js/FFEntryPoint.js [hunk 2]
```diff
@@ -51,9 +66,10 @@ if (app) {
                 {label:'Example Js Dialog', action:'exampleDialog', type:'COMMAND'}
         ]
     };
-    const props = Object.assign({}, defProps, app);
+    const props = Object.assign(defaults, app);
+    const viewer = app.template === Template.LCV ? LcViewer : FireflyViewer;
```
Comment (robyww) [inline src/firefly/js/FFEntryPoint.js]: should this line and the constants above be in a some sort of factory?

File: src/firefly/js/templates/lightcurve/LcResult.jsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        var {expanded, standard, closeable} = mode || {};
+        const content = {};
+        var visToolbar;
+        if (showImages) {
+            visToolbar = <VisToolbar key='res-vis-tb'/>;
+            content.imagePlot = (<TriViewImageSection key='res-tri-img'
+                                        closeable={closeable}
+                                        imageExpandedMode={expanded===LO_VIEW.images}
+                                        {...images}  />);
+        }
```
Comment (robyww) [inline src/firefly/js/templates/lightcurve/LcResult.jsx]: I don't think you want a TriViewImageSection here.  Must a MultiImageViewer.  If we want coverage and user added image in the LCViewer the we probably need to generalize TriViewImageSection to another component.

File: src/firefly/js/templates/lightcurve/LcManager.js [hunk 4]
```diff
@@ -0,0 +1,209 @@
+/*
```
Comment (robyww) [inline src/firefly/js/templates/lightcurve/LcManager.js]: LcManager is mostly a placeholder, right?

**Comment by robyww** [comment]
Is LayoutCntlr more general than just the tri-view? I know it is required for expanded but it is not clear if is for others.  Maybe I don't understand it well enough.  I am having trouble which what goes in LayoutCntlr.

**Comment by robyww** [comment]
should the triview stuff be moved to templates/tri-view?

**Comment by robyww** [comment]
Conclusions:

- The structure looks good.  It provides both flexibility and simplicity.
- I think we might want to have everything about a templates together: constant and code. Maybe we import a template object that has the constant and the React Element?
- I think I need to further separate the saga from the TriViewImageSection and rename it.
- I need to understand LayoutCntrl better, I have used with out enough understanding.

I think the only thing you should change for this ticket is the template organization (tri-view in templates directory), other stuff we can iterate on).

This is really good path forward.  
Review Complete.

File: src/firefly/html/lc.html (hunk 1 above)
Comment (loitly) [inline src/firefly/html/lc.html]: Yes, FireflyViewer is the default.  It's documented via jsdoc in FFEntryPoint.app.

File: src/firefly/js/FFEntryPoint.js (hunk 2 above)
Comment (loitly) [inline src/firefly/js/FFEntryPoint.js]: I was thinking just a private map.  I don't think anyone else need to know about this.

File: src/firefly/js/templates/lightcurve/LcResult.jsx (hunk 3 above)
Comment (loitly) [inline src/firefly/js/templates/lightcurve/LcResult.jsx]: I am sure you are right.  TriViewImageSection is just a placeholder.  I did not load the saga.  Beside the raw table, all of the other components are placeholders as well.

File: src/firefly/js/templates/lightcurve/LcManager.js (hunk 4 above)
Comment (loitly) [inline src/firefly/js/templates/lightcurve/LcManager.js]: Yes, it will be a complete rewrite.  There is a jira ticket for this.

**Pull request merged**